aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -102,6 +108,39 @@ void convert_indices_from_csr_to_coo_cuda(const Tensor& indices, const Tensor& c
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+template <typename scalar_t>
+void _apply_sparse_csr_linear_solve(
```

Comment:
Guarding with `USE_ROCM` should prevent the ROCm compilation from failing.
```cpp
#ifdef USE_ROCM
  TORCH_CHECK(
      false,
      "Calling torch.linalg.solve with sparse tensors requires compiling ",
      "PyTorch with CUDA and not supported in ROCm build.");
#else
// main code
#endif
```

Reply:
Done, thanks Ivan!